### PR TITLE
Add warnings about limited availability of versioning features

### PR DESCRIPTION
**Pull request by artonge**
Fix https://github.com/nextcloud/server/issues/37730

### CODE REVIEW

File: user_manual/files/version_control.rst [hunk 1]
```diff
@@ -36,6 +36,8 @@ versions until it meets the disk space limit again.
 Naming a version
 ----------------
 
+.. warning:: Naming a version is currently not enabled when group folders or S3 versioning is enabled.
```
Comment (szaimen) [inline user_manual/files/version_control.rst:39]:
```suggestion
.. warning:: Naming a version is currently not available when group folders or S3 versioning is enabled.
```

File: user_manual/files/version_control.rst [hunk 2]
```diff
@@ -46,4 +48,6 @@ When a version has a name, it will be excluded from the automatic expiration pro
 Deleting a version
 ------------------
 
+.. warning:: Deleting a version is currently not enabled when group folders or S3 versioning is enabled.
```
Comment (szaimen) [inline user_manual/files/version_control.rst:51]:
```suggestion
.. warning:: Deleting a version is currently not available when group folders or S3 versioning is enabled.
```

File: user_manual/files/version_control.rst [hunk 3]
```diff
@@ -46,4 +48,6 @@ When a version has a name, it will be excluded from the automatic expiration pro
 Deleting a version
 ------------------
 
+.. warning:: Deleting a version is currently not available when group folders or S3 versioning is enabled.
```
Comment (szaimen) [inline user_manual/files/version_control.rst:51]:
```suggestion
.. warning:: Deleting a version is currently not available when the group folders app or S3 versioning app is enabled.
```

File: user_manual/files/version_control.rst [hunk 4]
```diff
@@ -36,6 +36,8 @@ versions until it meets the disk space limit again.
 Naming a version
 ----------------
 
+.. warning:: Naming a version is currently not available when group folders or S3 versioning is enabled.
```
[the remark this answers is not in the document]
Comment (artonge) [reply user_manual/files/version_control.rst]: Adapted to go over the top

**Review by szaimen — APPROVED** [review]

**Pull request merged**